core/src/main/scala/org/apache/spark/Accumulable.scala

```diff
@@ -49,6 +48,7 @@ import org.apache.spark.util.Utils
  * @tparam R the full accumulated data (result type)
  * @tparam T partial data that can be added in
  */
+@deprecated("use AccumulatorV2", "2.0.0")
```

Comment:
yup we need to tweak the scala api a little bit. We can deprecate and create the Java APIs once the Scala done is finalized. I have a ticket for that.